packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -476,4 +529,117 @@ private void clearMethodCallAndResult() {
+    }
+    if (errorMessage != null) {
+      editor.putString(SHARED_PREFERENCE_ERROR_MESSAGE_KEY, errorMessage);
+    }
+    editor.apply();
+  }
+
+  private void cleanPendingResultPref() {
+    if (ImagePickerPlugin.getFilePref == null) {
+      return;
+    }
+    ImagePickerPlugin.getFilePref.edit().clear().apply();
+  }
+
+  private Map<String, String> resultMapFromPendingResult() {
+    if (ImagePickerPlugin.getFilePref == null) {
+      return new HashMap<>();
+    }
+    Map<String, String> resultMap = new HashMap<>();
+    Boolean hasData = false;
+    String filePath =
+        ImagePickerPlugin.getFilePref.getString(FLUTTER_IMAGE_PICKER_IMAGE_PATH_KEY, null);
+    if (filePath != null) {
+      String finalPath = getFinalImagePath(filePath);
+      resultMap.put("path", finalPath);
+      hasData = true;
+    }
+    String errorCode =
+        ImagePickerPlugin.getFilePref.getString(SHARED_PREFERENCE_ERROR_CODE_KEY, null);
+    if (errorCode != null) {
+      resultMap.put("errorCode", errorCode);
+      hasData = true;
+      String errorMessage =
+          ImagePickerPlugin.getFilePref.getString(SHARED_PREFERENCE_ERROR_MESSAGE_KEY, null);
+      if (errorMessage != null) {
+        resultMap.put("errorMessage", errorMessage);
+      }
+    }
+
+    if (hasData) {
+      String type = ImagePickerPlugin.getFilePref.getString(SHARED_PREFERENCE_TYPE_KEY, null);
+      if (type != null) {
+        resultMap.put("type", type);
+      }
+    }
+    return resultMap;
+  }
+
+  private Double getMaxWidth() {
+    if (ImagePickerPlugin.getFilePref == null
+        || !ImagePickerPlugin.getFilePref.contains(SHARED_PREFERENCE_MAX_WIDTH_KEY)) {
+      return null;
+    }
+    return Double.longBitsToDouble(
+        ImagePickerPlugin.getFilePref.getLong(SHARED_PREFERENCE_MAX_WIDTH_KEY, 0));
+  }
+
+  private Double getMaxHeight() {
+    if (ImagePickerPlugin.getFilePref == null
+        || !ImagePickerPlugin.getFilePref.contains(SHARED_PREFERENCE_MAX_HEIGHT_KEY)) {
+      return null;
+    }
+    return Double.longBitsToDouble(
+        ImagePickerPlugin.getFilePref.getLong(SHARED_PREFERENCE_MAX_HEIGHT_KEY, 0));
+  }
+
+  private void saveMaxDimension(Double maxWidth, Double maxHeight) {
+    if (ImagePickerPlugin.getFilePref != null) {
+      SharedPreferences.Editor editor = ImagePickerPlugin.getFilePref.edit();
+      if (maxWidth != null) {
+        editor.putLong(SHARED_PREFERENCE_MAX_WIDTH_KEY, Double.doubleToRawLongBits(maxWidth));
+      }
+      if (maxHeight != null) {
+        editor.putLong(SHARED_PREFERENCE_MAX_HEIGHT_KEY, Double.doubleToRawLongBits(maxHeight));
+      }
+      editor.apply();
+    }
+  }
+
+  private void setType(String type) {
+    if (ImagePickerPlugin.getFilePref == null) {
+      return;
+    }
+    ImagePickerPlugin.getFilePref.edit().putString(SHARED_PREFERENCE_TYPE_KEY, type).apply();
+  }
+
+  private void persistsPendingCameraMediaUriPath(String path) {
```

Comment:
nit: `persistPendingCameraMediaUriPath`. That way it matches the same tense as the rest of the methods.